api4/team.go

```diff
@@ -231,6 +232,55 @@ func patchTeam(c *Context, w http.ResponseWriter, r *http.Request) {
 	w.Write([]byte(patchedTeam.ToJson()))
 }
 
+func updateTeamPrivacy(c *Context, w http.ResponseWriter, r *http.Request) {
+	c.RequireTeamId()
+	if c.Err != nil {
+		return
+	}
+
+	c.RequirePrivacy()
+	if c.Err != nil {
+		return
+	}
+
+	auditRec := c.MakeAuditRecord("updateTeamPrivacy", audit.Fail)
+	defer c.LogAuditRec(auditRec)
+	auditRec.AddMeta("privacy", c.Params.Privacy)
+
+	if !c.App.SessionHasPermissionToTeam(*c.App.Session(), c.Params.TeamId, model.PERMISSION_MANAGE_TEAM) {
+		auditRec.AddMeta("team_id", c.Params.TeamId)
+		c.SetPermissionError(model.PERMISSION_MANAGE_TEAM)
+		return
+	}
+
+	if team, err := c.App.GetTeam(c.Params.TeamId); err == nil {
+		auditRec.AddMeta("team", team)
+	}
+
+	var teamType string
+	var openInvite bool
+	switch c.Params.Privacy {
+	case "public":
+		teamType = model.TEAM_OPEN
+		openInvite = true
+	case "private":
+		teamType = model.TEAM_INVITE
+		openInvite = false
+	default:
+		c.SetInvalidUrlParam("privacy")
+		return
+	}
+
+	err := c.App.UpdateTeamPrivacy(c.Params.TeamId, teamType, openInvite)
```

Comment:
Maybe `if err := ...; err != nil {}` form?

Reply:
I made the change.  TBH, I'm not a fan of that pattern when a single function is mixing both styles. Sometimes you need one of the vars elsewhere and must use `something, err := ...`, and then the mixing of style one after the other is harder to read IMO.